packages/firebase_admob/lib/firebase_admob.dart

```diff
@@ -82,6 +82,65 @@ class MobileAdTargetingInfo {
   }
 }
 
+// The types of ad sizes supported for banners.
+enum AdSizeType {
+  WidthAndHeight,
+  SmartBanner,
+}
+
+/// [AdSize] represents the size of a banner ad.
+class AdSize {
+  final int height;
+  final int width;
+  final AdSizeType sizeType;
+
+  /// Creates a typical width-by-height banner size. In most cases, it's easier
+  /// to use one of the named constructors like [AdSize.banner].
+  AdSize({this.width, this.height})
+      : assert(width > 0),
+        assert(height > 0),
+        sizeType = AdSizeType.WidthAndHeight;
+
+  /// Creates a standard banner (320x50) size.
+  AdSize.banner()
+      : width = 320,
+        height = 50,
+        sizeType = AdSizeType.WidthAndHeight;
+
+  /// Creates a large banner (320x100) size.
+  AdSize.largeBanner()
+      : width = 320,
+        height = 100,
+        sizeType = AdSizeType.WidthAndHeight;
+
+  /// Creates a medium rectangle (300x250) size.
+  AdSize.mediumRectangle()
+      : width = 300,
+        height = 250,
+        sizeType = AdSizeType.WidthAndHeight;
+
+  /// Creates a full banner (468x60) size.
+  AdSize.fullBanner()
+      : width = 468,
+        height = 60,
+        sizeType = AdSizeType.WidthAndHeight;
+
+  /// Creates a leaderboard (728x90) size.
+  AdSize.leaderboard()
+      : width = 728,
+        height = 90,
+        sizeType = AdSizeType.WidthAndHeight;
+
+  /// Creates a smart banner size. Smart banners are unique in that they will
+  /// automatically adjust their width to match that of the displaying device's
+  /// screen. Height also varies, and will be 32, 50, or 90 depending on how
+  /// tall the screen is.
+  AdSize.smartBanner()
```

Comment:
From the docs, it seems like there are only three valid height values (and you can't really specify the width?). You could either provide three constants or just provide a const constructor whose width could only be one of the 3 valid values.

Since the recommended practice is to base the banner's size on the screen size, you could also provide a `smartBannerFor(MediaQueryData)` constructor that DTRT.

Reply:
I think I need to revise this comment, since it clearly didn't impart the right info. :)

If you choose "smart banner" as your ad size, there are no other inputs you're allowed to give. You, as the publisher, do not determine how wide or high the ad is. The SDK will automatically set the width of the AdView to match the device width, and will set its height based on [a calculation involving device height](https://developers.google.com/admob/android/banner#smart_banners). So there's nothing more for the API to do here other than give the publisher a way to say "make me an AdSize object of the smart banner type."

Reply:
Sorry, I just didn't read this stuff carefully enough.

Reply:
You shouldn't have to, though, so it's good feedback.